spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/JdbcQueryCreator.java

```diff
@@ -267,7 +260,19 @@ private SelectBuilder.SelectJoin selectBuilder(Table table) {
 		SelectBuilder.SelectJoin baseSelect = selectBuilder.from(table);
 
 		for (Join join : joinTables) {
-			baseSelect = baseSelect.leftOuterJoin(join.joinTable).on(join.joinColumn).equals(join.parentId);
+
+			Condition condition = null;
+
+			for (int i = 0; i < join.joinColumns.size(); i++) {
```

Comment:
This looks familiar from what is in `SqlGenerator.Join`. Additionally, there's this strange contract of same list ordering and indices.

Reply:
Removed the duplication